### PR TITLE
[@types/theo] Theo 8.1 updates support

### PR DESCRIPTION
**Pull request by laitine**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
* Added all currently available formats
* Fixed bug with registerFormat which didn’t allow creation of custom formats like ’array.json'
* Added optional convertOptions
* Added fields to StyleMap
* Changed Immutable version to match Theo
* Loosen Transform and ValueTransform to string to allow registered custom values
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by petekp — APPROVED** [review]
@laitine Looks good! thanks for doing this 😃

File: types/theo/index.d.ts [hunk 1]
```diff
@@ -43,7 +49,7 @@ export type ValueTransform =
 export function convert(options: ConvertOptions): Promise<string>;
 export function convertSync(options: ConvertOptions): string;
 export function registerFormat(
-    name: Format,
+    name: string,
```
Comment (petekp) [inline types/theo/index.d.ts]: Good suggestion to open this up for custom formats 👍

File: types/theo/index.d.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
-// Type definitions for theo 8.0
+// Type definitions for theo 8.1
 // Project: https://github.com/salesforce-ux/theo
 // Definitions by: Pete Petrash <https://github.com/petekp>
+//                 Niko Laitinen <https://github.com/laitine>
```
Comment (petekp) [inline types/theo/index.d.ts]: 🙌

File: types/theo/index.d.ts (hunk 1 above)
Comment (petekp) [reply types/theo/index.d.ts]: Wonder if we'd want to loosen up on the `Transform` and `ValueTransform` types to make customization easier as well.

Comment (laitine) [reply types/theo/index.d.ts]: I was thinking about this too. I could do this too while at it. :-)

**Comment by laitine** [comment]
@petekp Thanks for putting up this type in the first place 🙇 👍 Ready for merge 💯

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Comment by petekp** [comment]
@isellsoap What was the build error out of curiosity — was it the TS 3.2 bump?

**Comment by laitine** [comment]
Needs a major version bump after all?

**Comment by petekp** [comment]
@laitine Not sure if this is what @isellsoap ran into, but we might consider changing the TS version back to `2.2` since the Theo 8.1 updates are still backwards compatible. Thoughts?

**Comment by laitine** [comment]
I think that's ok. If in case that was the reason why the build broke.

**Comment by petekp** [comment]
@laitine Cool, made a PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33040

**Comment by isellsoap** [comment]
@petekp The error was that my interface was not correct anymore, as the type got changed from `Transform` to `string`. So it was a compile error. Changing the API in that way should definitely be treated as a breaking change, or?

**Comment by petekp** [comment]
@isellsoap Apologies for that. I will dig in tonight and will attempt to reproduce and resolve that issue. In theory, those types being changed to `string` should still be valid as long as the parameter is a string, so it's unexpected that it would blow up—will take a closer look.

Ideally, these types would be flexible enough to allow for arbitrary/custom transform strings as well as providing autocomplete for Theo's built-in transforms. Need to do some more reading to see if that is possible without the types just flattening to `string`.

**Pull request merged**